modules/subprocess/src/main/scala/notebook/kernel/pfork/BetterFork.scala

```diff
@@ -110,12 +109,11 @@ class BetterFork[A <: ForkableProcess : reflect.ClassTag](config: Config,
 
     Future {
       log.info("Spawning %s".format(cmd.toString))
-
       // use environment because classpaths can be longer here than as a command line arg
-      val environment = System.getenv + ("CLASSPATH" -> (
-        sys.env.get("HADOOP_CONF_DIR").map(_ + ":").getOrElse("") +
-        sys.env.get("EXTRA_CLASSPATH").map(_ + ":").getOrElse("") +
-        classPathString))
+      val classpath = classPath ++ Array(
+        sys.env.get("HADOOP_CONF_DIR").getOrElse(""),
+        sys.env.get("EXTRA_CLASSPATH").getOrElse(""))
+      val environment = System.getenv + ("CLASSPATH" -> StringUtils.toString(classpath.toArray, File.pathSeparator))
```

Comment:
:+1:  for `File.pathSeparator`!

However, it would be better to avoid using `StringUtils` here I guess, this could be easily achieved like this:
```scala
classpath.mkString(File.pathSeparator)
```
my2¢